compiler.cpp

```diff
@@ -319,18 +319,16 @@ const Strat* ComputeStrategy(const Policy& node, std::unordered_map<const Policy
                     keys.push_back(x.keys[0]);
                 }
                 strats.push_back(MakeStrat(store, Strat::Type::MULTI, std::move(keys), node.k));
-            }
-            if (node.k > 1 && node.k < node.sub.size()) {
-                strats.push_back(MakeStrat(store, Strat::Type::THRESH, subs, node.k, (double)node.k / subs.size()));
-            }
-            if (node.k == 1 || node.k == node.sub.size()) {
+            } else if ((node.k == 1 && node.sub.size() < 10) || node.k == node.sub.size()) {
```

Comment:
Where is this limit 10 coming from?

Reply:
It does, but at least it results at some point :)

Reply:
In that case I'd prefer to fix that in other ways (e.g. have a simpler or/and strategy that doesn't consider all probabilities).

Reply:
Hmm ok so i think it was a bit clumsy for me to try to solve this in this PR since it gets outside the scope and was a hack for the specific `thresh` case but a decent solution would also encompass the existing nested-`or()`s case.

How about leaving this as a follow-up? This PR does not make the existing situation worse (but still opens up a new situation where this can be encountered..).

Reply:
Yeah, sure - I just noticed too late this PR was changing more than what I had expected it to do.